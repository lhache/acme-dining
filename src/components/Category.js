import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './Category.css'

class Category extends Component {
  render() {
    const { name } = this.props.category
    return (
      <div className="Category">
        <Link to={`/?category=${name}`} className='CategoryLink'>{name}</Link>
      </div>
    );
  }
}

export default Category
