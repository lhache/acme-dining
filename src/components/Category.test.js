import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { withApollo } from 'react-apollo';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Category from './Category';

configure({ adapter: new Adapter() });

const category = {
  name: 'name'
}

it('renders without crashing', () => {
  const wrapper = shallow(<Category category={category} />)
  expect(wrapper).toMatchSnapshot();
});

it('renders with proper props', () => {
  const wrapper = shallow(<Category category={category} />)
  const inst = wrapper.instance();
  expect(inst.props.category).toBe(category)
});
