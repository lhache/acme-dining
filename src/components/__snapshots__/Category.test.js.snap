// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Category
    category={
      Object {
        "name": "name",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="CategoryLink"
        replace={false}
        to="/?category=name"
      >
        name
      </Link>,
      "className": "Category",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": "name",
        "className": "CategoryLink",
        "replace": false,
        "to": "/?category=name",
      },
      "ref": null,
      "rendered": "name",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="CategoryLink"
          replace={false}
          to="/?category=name"
        >
          name
        </Link>,
        "className": "Category",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "name",
          "className": "CategoryLink",
          "replace": false,
          "to": "/?category=name",
        },
        "ref": null,
        "rendered": "name",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
